toolchain/check/testdata/class/adapter/adapt_copy.carbon

```diff
@@ -0,0 +1,477 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/class/adapter/adapt_copy.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/class/adapter/adapt_copy.carbon
+
+// --- adapt_copyable.carbon
+
+library "[[@TEST_NAME]]";
+
+class AdaptCopyable {
+  adapt i32;
+}
+
+// We can copy an adapter by copying its value if its initializing
+// representation is a copy of its value representation.
+fn F(c: AdaptCopyable) -> AdaptCopyable {
+  var d: AdaptCopyable = c;
+  return d;
+}
+
+fn InTuple(c: (AdaptCopyable, i32)) -> (AdaptCopyable, i32) {
+  var d: (AdaptCopyable, i32) = c;
+  return d;
+}
+
+// --- fail_todo_adapt_copyable_tuple.carbon
+
+library "[[@TEST_NAME]]";
+
+class AdaptTuple {
+  adapt (i32, i32);
+}
+
+// TODO: Support copying in this case too, by performing the corresponding copy
```

Comment:
If you don't have this in progress already, would it be okay for me to attempt to fix this TODO now?

Reply:
Yes, please do! Thanks!